ideas/282-deterministic-builds.md

```diff
@@ -0,0 +1,53 @@
+---
+id: 282-deterministic-builds
+title: Deterministic Mobile App Builds
+status: In Progress
+created: 2018-08-16
+category: infra
+lead-contributor: jakubgs
+contributors:
+exit-criteria: yes
+success-metrics: yes
+clear-roles: yes
+future-iterations: yes
+roles-needed:
+---
+
+## Preamble
+
+    Idea: #282-deterministic-builds
+    Title: Deterministic Mobile App Builds
+    Status: In Progress
+    Created: 2018-08-16
+
+The purpose of achieving deterministic builds is encuring security of Status software. Deterministic builds depend on ensuring stable inputs, ensuring stable outputs, and capturing as little of the build environment as possible.
+
+In simple terms, deterministic builds mean the same commit generates exactly the same software package(s) regardless of where and when it is built. This is achieved by controlling all possible variables in a build, which includes controlling your build environment as well.
+
+## Objectives
+
+- status-go builds are deterministic
+- status-react builds are deterministic
+
+## Key Results
+
+- Anyone can build a commit on their own machines and verify the result by comparing against our builds
+- We can upload Android builds to F-Droid which requires deterministic builds
+
+## Timeline / Checkpoints
+
+__TODO__: Not sure?
```

Comment:
What's something reasonable that can be done in first iteration? Perhaps something like find out current state of each sub package/component and what biggest blockers are. This way the work can be distributed, so someone might look into JS vendoring, and someone else figuring out how to get out timestamps from some ObjC linked monster config file.

Reply:
That sounds reasonable, thanks.